pkg/datapath/tables/node_address.go

```diff
@@ -294,20 +294,13 @@ func (n *nodeAddressController) getAddressesFromDevice(dev *Device) (addrs sets.
 		return
 	}
 
+	addressScopeMax := n.AddressScopeMax
+
 	if dev.Name == defaults.HostDevice {
 		// If AddressScopeMax is a scope more broad (numerically less than) than SCOPE_LINK then
 		// include all addresses at SCOPE_LINK which are assigned to the Cilium host device.
 		if n.AddressScopeMax < unix.RT_SCOPE_LINK {
-			for _, addr := range sortedAddresses(dev.Addrs) {
-				if addr.Scope == unix.RT_SCOPE_LINK {
-					addrs.Insert(NodeAddress{
-						Addr:       addr.Addr,
-						NodePort:   false,
-						Primary:    !addr.Secondary,
-						DeviceName: dev.Name,
-					})
-				}
-			}
+			addressScopeMax = unix.RT_SCOPE_LINK
```

Comment:
doesn't this change behaviour for addrs on `cilium_host` with `RT_SCOPE_SITE` now?

SITE is < LINK - old code only added addresses with `LINK`, but AFAICT a SITE scoped addr would be added now

Reply:
The old code still added all addresses based on the max scope. But indeed, this is different in that it might added addresses that are between the max scope and link that weren't added before. I'll change it to keep the semantics.